js/src/modal.js

```diff
@@ -240,6 +247,9 @@ class Modal extends BaseComponent {
     }
 
     this._element.classList.add(CLASS_NAME_SHOW)
+    if (this._hasCustomRoot()) {
+      this._element.classList.add(CLASS_NAME_MODAL_CUSTOM_ROOT)
```

Comment:
Maybe you can avoid these custom root classes and checks, by simple using a css selector like `.modal-open:not(body)`

Reply:
Good idea, amended.